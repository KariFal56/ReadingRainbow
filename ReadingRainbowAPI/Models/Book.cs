using System.Collections.Generic;

namespace ReadingRainbowAPI.Models
{
    public class Book : Neo4jEntity
    {
        public string Id {get; set;}
        public string BookInformationLink {get; set;}
        public string Title {get; set;}
        public List<string> Authors = new List<string>();
        public string Thumbnail {get; set;}
        public string SmallThumbnail {get; set;}
        public string PublishDate {get; set;}
        public string NumberPages {get; set;}
        public string Description {get; set;}
        public string ISBN_10 {get; set;}      
        public string ISBN_13 {get; set;}   
        public List<string> Cateogries = new List<string>();

    }   
}