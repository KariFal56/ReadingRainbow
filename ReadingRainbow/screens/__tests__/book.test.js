import React from 'react';
import { act } from 'react-test-renderer';
import Book from '../book';
import { NavigationContainer } from "@react-navigation/native"
import { createStackNavigator } from '@react-navigation/stack';
import { render } from '@testing-library/react-native';
// import MockedNavigator from '../mocked_navigator';

const Stack = createStackNavigator();

test('two plus three is five', () => {
    expect(2 + 3).toBe(5);
});

describe('Book renders correctly', () => {
    it ('should match snapshot', async () => {
        const result = render(
            <NavigationContainer>
                <Stack.Screen name="Book" component={Book} />
            </NavigationContainer>
        );
        await act(async () => { expect(result).toMatchSnapshot(); })
    })
})

// it('Book renders correctly', () => {
//     const {toJSON} = render(
//       <MockedNavigator component={Book} />
//     );
//     expect(toJSON()).toMatchSnapshot();
// });

// it('Book renders correctly', () => {
//     let tree;

//     act(() => {
//         tree = create(
//             <MockedNavigator component={Book} />
//         );
//     });
//     expect(tree).toMatchSnapshot();
// });

