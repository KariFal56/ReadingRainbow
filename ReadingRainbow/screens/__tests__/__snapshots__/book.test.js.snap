// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Book renders correctly should match snapshot 1`] = `null`;
